core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -247,11 +252,17 @@ private boolean isEndpointGatewayEnabled(EnvironmentNetworkResponse network) {
             knoxTargetGroup.setType(TargetGroupType.KNOX);
             knoxTargetGroup.setInstanceGroups(knoxGatewayInstanceGroups);
             TargetGroup finalKnoxTargetGroup = knoxTargetGroup;
-            knoxGatewayInstanceGroups.forEach(ig -> ig.addTargetGroup(finalKnoxTargetGroup));
+            knoxGatewayInstanceGroups.forEach(ig -> addTargetToInstanceGroup(ig, finalKnoxTargetGroup));
         }
         return Optional.ofNullable(knoxTargetGroup);
     }
 
+    private void addTargetToInstanceGroup(InstanceGroup instanceGroup, TargetGroup targetGroup) {
```

Comment:
This is called from a cycle I would think about having one proper named query to fetch the necessary lines from the two affected tables.

Reply:
I'm not sure what you mean. Can you elaborate?

Reply:
I've changed the logic here so we're no longer fetching from the database, and have also removed the Stack database fetches from the various handles. The exception is the `CreateLoadBalancerEntityHandler`, because the purpose of that handler is to fetch the database entities and update them. But after that fetch, the stack is passed between handlers instead of making another database call.